jdaviz/configs/imviz/plugins/footprints/footprints.py

```diff
@@ -372,10 +418,20 @@ def overlay_regions(self):
         callable_kwargs = {k: getattr(self, k)
                            for k in ('ra', 'dec', 'pa', 'v2_offset', 'v3_offset')}
 
-        regs = preset_regions.jwst_footprint(self.preset_selected, **callable_kwargs)
+        if self.preset_selected == 'From File...':
+            # we need to cache these locally in order to support multiple files/regions between
+            # different overlay entries all selecting From File...
+            overlay = self._overlays.get(self.overlay_selected, {})
+            if 'regions' not in overlay and isinstance(self.preset.selected_obj, regions.Regions):
+                overlay['regions'] = self.preset.selected_obj
+            regs = overlay.get('regions')

```

Comment:
Is this returning `None` under some condition that caused Ricky's usage to crash?

Reply:
yep, that was it.  Needs to default to `[]` not `None`.  Fixed now!